### PR TITLE
Add transient staging library.

### PR DESCRIPTION
**Pull request by robertzk**
This will prevent issues of the form.

```
ERROR: ‘foo’ is not a legal package name
```

### CODE REVIEW

File: R/zzz.R [hunk 1]
```diff
@@ -6,7 +6,16 @@ set_transient_library <- function() {
...
   if (!file.exists(dir)) dir.create(dir, FALSE, TRUE)
   .lockbox_env$old_dir <- .libPaths()
-  .libPaths(c(dir, .libPaths()))
+
+  # We add one final library path: a transient staging library
+  # that is used to copy over installed packages to the vanilla
+  # library.
+  transient_staging_path <- transientStagingPath()
+  if (!file.exists(transient_staging_path)) {
+    dir.create(transient_staging_path, FALSE, TRUE)
```
Comment (peterhurford) [inline R/zzz.R]: You should use variable names for the `FALSE` and `TRUE` for readibilty.

File: R/zzz.R [hunk 2]
```diff
@@ -6,7 +6,16 @@ set_transient_library <- function() {
...
+    unlink(transient_staging_path, TRUE, TRUE)
```
Comment (peterhurford) [inline R/zzz.R:15]: Ditto.

**Comment by peterhurford** [comment]
:+1: just added a suggestion for readibility.

**Comment by kirillseva** [comment]
:+1:

**Pull request merged**